parlai/tasks/wrapper/agents.py

```diff
@@ -144,3 +131,33 @@ def shutdown(self):
 
     def update_counters(self):
         self.task.update_counters()
+
+
+class LabelToTextTeacher(AbstractWrapperTeacher):
+    """
+    Teacher that will shift message['labels'][0] into message['text'] for whatever task
+    is specified with --wrapper-task.
+
+    Because the dialogue history is effectively overwritten by this action, all episodes
+    will be flattened into one example each.
+    """
+
+    def __init__(self, opt: Opt, shared=None):
+        super().__init__(opt, shared)
+
+    def act(self):
+        """
+        Act on the previous observation.
+        """
+        act = self.task.act()
+        new_act = copy.deepcopy(act)
+        if 'labels' in act or 'eval_labels' in act:
+            labels_type = 'labels' if 'labels' in act else 'eval_labels'
+            labels = act[labels_type]
+            assert len(labels) == 1
```

Comment:
why this assert here? this is not universally true in ParlAI, so if required we should add a more descriptive error message

Reply:
Okay cool - this was probably originally my line, because I've never seen a case in which there's more than one label. But yes, generalizing this seems like a good idea, then - it's good to future-proof this teacher

Reply:
Yeah, this teacher is @EricMichaelSmith code. Should I add the descriptive message stating it was created for the specific task you created it for and that it can be comment out if needed?

Reply:
hmm - maybe you could do
```
if len(labels) != 1:
    raise ValueError('LabelToTextTeacher can only be used with one label!')
```
and then in the future, if there's a need to use this teacher with more than one label, we can figure out what logic we want to use for that then